consensus/src/round_manager.rs

```diff
@@ -464,12 +448,22 @@ impl RoundManager {
         Ok(())
     }
 
-    /// This function processes a proposal that was chosen as a representative of its round:
-    /// 1. Add it to a block store.
-    /// 2. Try to vote for it following the safety rules.
-    /// 3. In case a validator chooses to vote, send the vote to the representatives at the next
-    /// position.
-    async fn process_proposed_block(&mut self, proposal: Block) -> Result<()> {
+    /// This function processes a proposal for the current round:
+    /// 1. Filter if it's proposed by valid proposer.
+    /// 2. Execute and add it to a block store.
+    /// 3. Try to vote for it following the safety rules.
+    /// 4. In case a validator chooses to vote, send the vote to the representatives at the next
+    /// round.
+    async fn process_proposal(&mut self, proposal: Block) -> Result<()> {
+        ensure!(
+            self.proposer_election.is_valid_proposal(&proposal),
+            "[RoundManager] Proposer {} for block {} is not a valid proposer for this round",
+            proposal
+                .author()
+                .expect("Proposal should be verified having an author"),
```

Comment:
could you elaborate so I don't have to find this? it doesn't seem like there was a delta from a previous PR.

Reply:
it's the same as the proposal_msg.proposer() since we only have proposal here